src/main/java/rs117/hd/HdPluginConfig.java

```diff
@@ -271,7 +271,7 @@ default LevelOfDetail levelOfDetail()
 	@ConfigItem(
 		keyName = "maxDynamicLights",
 		name = "Dynamic Lights",
-		description = "The maximum number of dynamic lights visible at one time. Reducing this will improve performance.",
+		description = "The maximum number of dynamic lights visible at one time. GPU-heavy in certain scenes.",
```

Comment:
I have not heard of anything functioning that way, do you have any source so that i can learn about it? Thanks

Reply:
I've been trying to look into it further, but it's hard to find any good resources on it, and I don't have first-hand experience with an old enough GPU. I've barely scratched the surface personally.

I think with the current requirements of the plugin (Shader Model 5.0), we don't really have to even consider it. It used to be that dynamic looping/branching (with a non-constant condition, i.e. number of lights) *wasn't allowed at all* before Shader Model 3.0 (which is ages ago, apparently around DX9 & GL2.1). Presumably there could've been some hiccups with optimization since then, but that's only speculation on my part. I would hope that by Shader Model 4.0 and even 5.0, any optimization issues around dynamic looping should have been resolved by all vendors.

There may still be room for optimization by unrolling the loop, which would make `MAX_LIGHTS` of 100 produce 100 iterations, and 25 produce 25, regardless of how many lights are currently in the scene. However, with the current implementation, there's no unrolling being done (it's doing actual looping by branching), since the compiler doesn't know what the upper bounds are for the loop (unless it's smarter than I think). Changing it to a constant number of lights might also just make performance worse if modern GPUs optimize loops well. It's a bit of a toss-up :shrug: